joomla/joomla-cms
# [4.0] plg_system_debug - Add option to select storage method

Pull Request for Issue #22137 .

### Summary of Changes

This will add an option to select a storage type for the debug plugin.

Currently only two options are supported:

1. File - Debug information will be written to disk in json format.
1. None

### Testing Instructions

1. Activate debug in global configuration
1. Switch the storage type in plugin configuration and confirm that files are written - or not.

![storage-setting](https://user-images.githubusercontent.com/33978/45582911-130fe900-b87e-11e8-8076-7dc176caa71c.png)


### Expected result

Ability to Set the storage type to 'None'

### Actual result

Files are always written - this might be a security issue.

### Documentation Changes Required

Yes.
Add words to description of #20380 (TBD)

close #22137
